lib/Driver/ToolChains/Flang.cpp

```diff
@@ -57,8 +57,7 @@ void FlangFrontend::ConstructJob(Compilation &C, const JobAction &JA,
   // Check file type sanity
   assert(types::isFortran(InputType) && "Can only accept Fortran");
 
-  if (Args.hasArg(options::OPT_fsyntax_only) ||
-      Args.hasArg(options::OPT_E)) {
+  if (Args.hasArg(options::OPT_fsyntax_only)) {
```

Comment:
These changes will mean that `Stem` contains garbage when it's later used in the command line.